### PR TITLE
Add docs for buildpackages

### LINKED ISSUE #107: Document buildpackages
- [ ] `create-package`
 - [ ] `create-builder` (esp. `builder.toml`)
 - [ ] `build`

### PR DESCRIPTION
**Pull request by natalieparellano**
Resolves #107

### CODE REVIEW

File: content/docs/app-developer-guide/specific-buildpacks.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ The `--buildpack` parameter can be one of the following:
 - path to a directory<sup><small>†</small></sup>, `tar` file, or `tgz` file
 - URL to a `tar` or `tgz` file
 - buildpack located in a builder, in the form of `<id>[@<version>]`<sup><small>‡</small></sup>
+- registry location
```
Comment (jromero) [inline content/docs/app-developer-guide/specific-buildpacks.md]: This should be more like:

```
- image name
```

File: content/docs/buildpack-author-guide/create-buildpack/make-buildpack-configurable.md [hunk 2]
```diff
@@ -128,3 +128,5 @@ Now that you've finished your buildpack, how about extending it? Try:
 - Caching the downloaded Ruby version
```
Comment (jromero) [inline content/docs/buildpack-author-guide/create-buildpack/make-buildpack-configurable.md]: Given that we have a follow up step I don't think this section is applicable anymore or maybe we can add packaging as a link in going further.

File: content/docs/buildpack-author-guide/create-buildpack/packaging.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
++++
+title="Packaging your buildpack for distribution"
+weight=408
+creatordisplayname = "Natalie Arellano"
```
Comment (jromero) [inline content/docs/buildpack-author-guide/create-buildpack/packaging.md]: We've stepped away from adding this metadata.

File: content/docs/concepts/components/buildpack.md [hunk 4]
```diff
@@ -39,6 +39,9 @@ During build the buildpacks contribute to the final application image. This cont
 some environment variables within the image, creating a layer containing a binary (e.g: node, python, or ruby), or 
 adding app dependencies (e.g: running `npm install`, `pip install -r requirements.txt`, or `bundle install`).
 
+## Distribution
+
+Buildpacks can be **packaged** as OCI images on an image registry or Docker daemon. This includes meta-buildpacks.
```
Comment (jromero) [inline content/docs/concepts/components/buildpack.md]: We should create a separate page under `Buildpack Author` that we can link to from here. Similar to content/docs/buildpack-author-guide/create-buildpack/packaging.md but isolated. Potentially using samples repo.

File: content/docs/reference/builder-config.md [hunk 5]
```diff
@@ -28,6 +28,23 @@ A [builder][builder] configuration schema is as follows:
...
+  - **`version`** _(string, optional)_
+    <br>
+    as defined above.
+
+  - **`image`** _(string, required)_
+    <br>
+    A registry location (if no registry host is specified in the image name, DockerHub is assumed).
+
 - #### `order` _(list, required)_
   <br>
```
Comment (jromero) [inline content/docs/reference/builder-config.md]: We should add a reference doc for `package.toml` as part of this story similar to this page.

File: content/docs/buildpack-author-guide/create-buildpack/make-buildpack-configurable.md (hunk 2 above)
Comment (natalieparellano) [reply content/docs/buildpack-author-guide/create-buildpack/make-buildpack-configurable.md]: Removed

File: content/docs/buildpack-author-guide/create-buildpack/packaging.md (hunk 3 above)
Comment (natalieparellano) [reply content/docs/buildpack-author-guide/create-buildpack/packaging.md]: Should I remove the entire metadata block? Or just specific fields... this data appears to be present in all of the sibling files.

File: content/docs/reference/builder-config.md (hunk 5 above)
Comment (natalieparellano) [reply content/docs/reference/builder-config.md]: Adding this.

File: content/docs/concepts/components/buildpack.md (hunk 4 above)
Comment (natalieparellano) [reply content/docs/concepts/components/buildpack.md]: Just pushed a change that adds a reference page for the package config. I started to write the step-by-step guide for a buildpack author but it felt a bit heavy-handed given that it's really just two steps (write the toml and run the pack command) and the intricacies of writing the toml are covered in the reference page. Happy to add though, if you really think it's required.

**Comment by jromero** [comment]
_Blocked until release of [pack 0.9.0](https://github.com/buildpacks/pack/milestone/13)_

**Comment by jromero** [comment]
_Blocked until release of [pack 0.9.0](https://github.com/buildpacks/pack/milestone/13)_

**Pull request merged**